lib/routes/utils/request_helper.js

```diff
@@ -80,8 +80,56 @@ function shouldSendVerifyAccountEmail(account, request) {
   return (!context || !!sendEmailIfUnverified) && !account.emailVerified
 }
 
+/**
+ * Returns whether or not to use signin unblock feature on a request.
+ *
+ * @param account
+ * @param config
+ * @param request
+ * @returns {boolean}
+ */
+function shouldEnableSigninUnblock(account, config, request) {
```

Comment:
Also over in #1470, I've migrated these feature predicates to a dedicated module `lib/features.js`. No need to include that here, just noting it as a reminder that we should move this one over after that PR is merged.